src/main/org/nlogo/prim/_breedhere.scala

```diff
@@ -0,0 +1,30 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.prim
+
+import org.nlogo.agent.{ AgentSet, ArrayAgentSet, Patch, Turtle }
+import org.nlogo.api.Syntax
+import org.nlogo.nvm.{ Context, Reporter }
+
+class _breedhere(breedName: String) extends Reporter {
+  override def syntax: Syntax = Syntax.reporterSyntax(Syntax.TurtlesetType, "-TP-")
+
+  override def toString: String = s"${super.toString}:$breedName"
+
+  override def report(context: Context) = report_1(context)
+
+  def report_1(context: Context): AgentSet = {
+    val patch = context.agent match {
+      case t: Turtle => t.getPatchHere
+      case p: Patch  => p
+    }
+    val breed = world.getBreed(breedName)
+    val agentSet = new ArrayAgentSet(classOf[Turtle], patch.turtleCount, false, world)
+    val itr = patch.turtlesHere.iterator
+    while (itr.hasNext) { val turtle = itr.next()
```

Comment:
It makes sense to me that you might want to write it like this, and I agree that it isn't hard to get used to. The problem is that it is not idiomatic scala. Although anyone *can* get used to it, they *have to* get used to it, and it can be a little surprising. `grep`-ing through both our codebase and the scala source, this pattern isn't used unless the `while` is on a single line, for instance:
```Scala
while (i.hasNext) { val a = i.next(); f(a) }
```
The person has to remember one idiom to use in one section of the code and a different idiom in a different section.

If you would like to convert any of the while loops to the above single-line format, the `val` can stay on the same line, otherwise it needs its own line.